### PR TITLE
Reduce the use of `async Task` where it's redundant

### PR DESCRIPTION
**Pull request by itsWindows11**
**Resolved / Related Issues**
- Related #9871
- Related #4180

**Details of Changes**
- Made some wrapper methods return the task directly rather than awaiting it.
- Most async methods with one async call at the end should have its task returned directly.
- Return `Task.CompletedTask` or `Task.FromResult<T>` when there's no async task to return.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

**Comment by yaichenbaum** [comment]
fyi @hez2010

**Comment by yaichenbaum** [comment]
Is this a performance improvement or is it more for code quality?

**Comment by hez2010** [comment]
The performance impact is minor. But it's better to have.

**Comment by yaichenbaum** [comment]
@itsWindows11 can you resolve the merge conflicts?

**Comment by hez2010** [comment]
> Most async methods with one async call at the end should have its task returned directly.

I'm not sure if it worth. We do can reduce a state machine by doing this, but it may have harmful impact on diagnostics (stacktrace).

**Comment by itsWindows11** [comment]
> > Most async methods with one async call at the end should have its task returned directly.
> 
> I'm not sure if it worth. We do can reduce a state machine by doing this, but it may have harmful impact on diagnostics (stacktrace). But we can try and see the result of this change. 

So should I keep it or remove it?

**Comment by itsWindows11** [comment]
> @itsWindows11 can you resolve the merge conflicts?

Currently busy on another project, will get into it tomorrow.

**Comment by hez2010** [comment]
> So should I keep it or remove it?

Let's keep it for now.

**Comment by itsWindows11** [comment]
@yaichenbaum PR should be ready for review now!

File: src/Files.FullTrust/MessageHandlers/FileOperationsHandler.cs [hunk 1]
```diff
@@ -35,15 +35,12 @@ public void Initialize(PipeStream connection)
             progressHandler = new ProgressHandler(connection);
         }
 
-        public async Task ParseArgumentsAsync(PipeStream connection, Dictionary<string, JsonElement> message, string arguments)
-        {
-            switch (arguments)
+        public Task ParseArgumentsAsync(PipeStream connection, Dictionary<string, object> message, string arguments)
```
Comment (hez2010) [inline src/Files.FullTrust/MessageHandlers/FileOperationsHandler.cs:38]: I think this line may have some merging issue. The message should be `Dictionary<string, JsonElement>`.

File: src/Files.FullTrust/MessageHandlers/LibrariesHandler.cs [hunk 2]
```diff
@@ -48,15 +48,12 @@ private void StartLibrariesWatcher()
             librariesWatcher.EnableRaisingEvents = true;
         }
 
-        public async Task ParseArgumentsAsync(PipeStream connection, Dictionary<string, JsonElement> message, string arguments)
-        {
-            switch (arguments)
+        public Task ParseArgumentsAsync(PipeStream connection, Dictionary<string, object> message, string arguments)
```
Comment (hez2010) [inline src/Files.FullTrust/MessageHandlers/LibrariesHandler.cs:51]: I think this line may have some merging issue. The message should be `Dictionary<string, JsonElement>`.

File: src/Files.FullTrust/MessageHandlers/DriveHandler.cs [hunk 3]
```diff
@@ -16,16 +16,18 @@ public class DriveHandler : Disposable, IMessageHandler
 
         public void Initialize(PipeStream connection) {}
 
-        public async Task ParseArgumentsAsync(PipeStream connection, Dictionary<string, JsonElement> message, string arguments)
+        public Task ParseArgumentsAsync(PipeStream connection, Dictionary<string, object> message, string arguments)
```
Comment (hez2010) [inline src/Files.FullTrust/MessageHandlers/DriveHandler.cs:19]: I think this line may have some merging issue. The message should be `Dictionary<string, JsonElement>`.

**Review by hez2010 — APPROVED** [review]
Otherwise LTGM.
Can you resolve the merge conflicts?

File: src/Files.FullTrust/MessageHandlers/DriveHandler.cs (hunk 3 above)
Comment (itsWindows11) [reply src/Files.FullTrust/MessageHandlers/DriveHandler.cs]: DriveHandler is supposed to be removed in the upstream so I merged the changes in the latest commit.

File: src/Files.FullTrust/MessageHandlers/LibrariesHandler.cs (hunk 2 above)
Comment (itsWindows11) [reply src/Files.FullTrust/MessageHandlers/LibrariesHandler.cs]: Oh, I think that was why pipeline builds were failing.

File: src/Files.FullTrust/MessageHandlers/FileOperationsHandler.cs (hunk 1 above)
Comment (itsWindows11) [reply src/Files.FullTrust/MessageHandlers/FileOperationsHandler.cs]: Oh, I think that was why pipeline builds were failing.

**Review by hez2010 — APPROVED** [review]
LGTM. Thank you.

**Comment by yaichenbaum** [comment]
Can you resolve the merge conflicts? Spacing should be set to tabs to avoid this happening again.

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**